api_gateway/src/main/kotlin/org/thechance/api_gateway/endpoints/utils/Extentions.kt

```diff
@@ -97,4 +97,22 @@ suspend inline fun <reified T> PipelineContext<Unit, ApplicationCall>.receiveMul
         part.dispose()
     }
     return MultipartDto(data = data!!, image = fileBytes)
+}
+
+fun String?.toListOfIntOrNull(): List<Int>? {
+    return if (!this.isNullOrBlank()) {
+        val integerStrings = this.replace("[","").replace("]","").split(",")
+        integerStrings.mapNotNull { it.toIntOrNull() }
+    } else {
+        null
+    }
+}
+
+fun String?.toListOfStringOrNull(): List<String>? {
+    return if (!this.isNullOrBlank()) {
+        val integerStrings = this.replace("[","").replace("]","").split(",")
+        integerStrings.map { it.trim() }
+    } else {
+        null
+    }
```

Comment:
try to use takeIf instead of if statement it's more cleaner
```suggestion
fun String?.toListOfStringOrNull(): List<String>? {
    return this.takeIf { !it.isNullOrBlank() }?.run {
        val integerStrings = this.replace("[", "").replace("]", "").split(",")
        integerStrings.map(String::trim)
    }
}
```